kernel/src/scan/mod.rs

```diff
@@ -269,6 +270,45 @@ impl Scan {
             let read_results =
                 parquet_handler.read_parquet_files(&[meta], self.physical_schema.clone(), None)?;
 
+            // The rest of the kernel expects variants to be reconstructed, so they must be
+            // reconstructed immediately after scanning the parquet files.
+            let variant_coalesce_expr = match &self.snapshot.protocol().reader_features {
+                Some(reader_features) => {
+                    // TODO(r.chen): Only check this if the variantType feature is enabled.
+                    // if reader_features.contains(&"variantType-preview".to_string())
+                    let all_fields_expr = 
+                        self.physical_schema
+                            .fields
+                            .iter()
+                            .map(|(_, field)| {
+                                Expression::variant_coalesce(Expression::column(
+                                    field.name.clone(),
+                                ))
```

Comment:
aside: (attn @nicklan) -- The blanket implementation of [ToString](https://doc.rust-lang.org/std/string/trait.ToString.html) trait will make a copy of even an owned string, because it's implemented in terms of `Display`. Maybe that's not quite what we want? 

Maybe these methods should take `impl Into<String>` instead? Technically, they're not equivalent (can't have an `impl<T: IntoString> From<T> for String` because it would conflict with the blanket `impl<T> From<T> for T`). That said, `String` has From<T>` for all the "obvious" types like `&str` and `&String`, so maybe we can get away with it in practice?